src/DeviceInterface/InterfacePrototype.js

```diff
@@ -70,6 +71,9 @@ class InterfacePrototype {
     /** @type {import("../../packages/device-api").DeviceApi} */
     deviceApi
 
+    /** @type {ReturnType<import("../locales/strings").getTranslator>} */
+    t = () => {}
```

Comment:
I think you could call your factory function here immediately, passing `this.settings` -> then in the closure you return you can access `settings.locale` when required.

that would prevent this empty function with overriden type, and will also prevent you from needing to re-assign `t` later in this file.